pkg/uhttp/wrapper.go

```diff
@@ -296,25 +298,26 @@ func (c *BaseHttpClient) Do(req *http.Request, options ...DoOption) (*http.Respo
 
 	switch resp.StatusCode {
 	case http.StatusRequestTimeout:
-		return resp, GRPCWrap(codes.DeadlineExceeded, resp, optErrs...)
+		return resp, WrapErrorsWithRateLimitInfo(codes.DeadlineExceeded, resp, optErrs...)
 	case http.StatusTooManyRequests, http.StatusServiceUnavailable:
-		return resp, GRPCWrap(codes.Unavailable, resp, optErrs...)
+		return resp, WrapErrorsWithRateLimitInfo(codes.Unavailable, resp, optErrs...)
 	case http.StatusNotFound:
-		return resp, GRPCWrap(codes.NotFound, resp, optErrs...)
+		return resp, WrapErrorsWithRateLimitInfo(codes.NotFound, resp, optErrs...)
 	case http.StatusUnauthorized:
-		return resp, GRPCWrap(codes.Unauthenticated, resp, optErrs...)
+		return resp, WrapErrorsWithRateLimitInfo(codes.Unauthenticated, resp, optErrs...)
 	case http.StatusForbidden:
-		return resp, GRPCWrap(codes.PermissionDenied, resp, optErrs...)
+		return resp, WrapErrorsWithRateLimitInfo(codes.PermissionDenied, resp, optErrs...)
 	case http.StatusNotImplemented:
-		return resp, GRPCWrap(codes.Unimplemented, resp, optErrs...)
+		return resp, WrapErrorsWithRateLimitInfo(codes.Unimplemented, resp, optErrs...)
 	}
 
 	if resp.StatusCode >= 500 && resp.StatusCode <= 599 {
-		return resp, GRPCWrap(codes.Unavailable, resp, optErrs...)
+		return resp, WrapErrorsWithRateLimitInfo(codes.Unavailable, resp, optErrs...)
 	}
 
 	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
-		return resp, GRPCWrap(codes.Unknown, resp, append(optErrs, fmt.Errorf("unexpected status code: %d", resp.StatusCode))...)
+		// TODO: add opterrs here
```

Comment:
This was TODONE in the previous code. I think you just want to rename the function call instead of going back to the old arguments.